src/sequential/sequential.cpp

```diff
@@ -0,0 +1,275 @@
+#include "sequential.hpp"
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// HELPER CLASSES
+////////////////////////////////////////////////////////////////////////////////
+
+bool
+SequentialBucket::is_empty() {
+  LOG_TRACE("Enter");
+  return this->offset == SIZE_MAX;
```

Comment:
This is the Pythonista in me, but I like everything to be explicit. As a novice C++ programmer, I don't want to be debugging at midnight wondering which object is being referenced.

I understand most people don't use it, but I can't help but think the benefits of being explicit will eventually outweigh the costs.

P.S. I know the common way is to have "var_" refer to the class variables.

Reply:
Yeah that's fair. The `var_` thing looks ugly with POD types where all members are public though. I'm ok with `this` as well, so long as we're consistent and use it everywhere.

Reply:
Some more food for thought: the main argument against `this` is that it only helps with readability if your code structure was sub-optimal to begin with. E.g. your function is too long, so it has too many variables. Or e.g., you're using it to disambiguate between a parameter and a member, which implies you're probably trying to write a setter function for a POD type:
```c++
struct Coordinate {
  int x;
  int y;

  void set_x(int x) {
    this->x = x;
  }
};
```

Reply:
What is your preferred way to create a setter function?

Reply:
Imo setter functions are only necessary when you need to enforce some kind of invariant, which implies you have logic beyond just setting the value of a member variable.

If you don't have invariants to enforce, then a simple member assignment is enough. Otherwise, a setter is just extra code to be maintained.